src/drivers/imu/analog_devices/adis16507/ADIS16507.cpp

```diff
@@ -234,196 +285,154 @@ void ADIS16507::RunImpl()
-				//  Gyroscope (Y-Axis) 1.51 ms
-				//  Gyroscope (Z-Axis) 1.29 ms
-				const uint64_t gyro_group_delay_us = (1'510 + 1'510 + 1'290) / 3;
-				_px4_gyro.update(timestamp_sample - gyro_group_delay_us, gyro_x, gyro_y, gyro_z);
+				return;
+			}
 
-				success = true;
+			_last_data_cntr = buffer.data_cntr;
 
-				if (_failure_count > 0) {
-					_failure_count--;
-				}
+			uint16_t checksum = 0;
+			uint8_t *checksum_helper = (uint8_t *)&buffer.diag_stat;
 
-			} else {
-				perf_count(_bad_transfer_perf);
+			for (int i = 0; i < 18; i++) {
+				checksum += checksum_helper[i];
 			}
 
-			if (!success) {
-				_failure_count++;
-
-				// full reset if things are failing consistently
-				if (_failure_count > 10) {
-					Reset();
-					return;
-				}
+			if (buffer.checksum != checksum) {
+				perf_count(_bad_checksum_perf);
+				// Don't publish if checksum fails
+				return;
 			}
 
-			if (!success || hrt_elapsed_time(&_last_config_check_timestamp) > 100_ms) {
-				// check configuration registers periodically or immediately following any failure
-				if (RegisterCheck(_register_cfg[_checked_register])) {
-					_last_config_check_timestamp = now;
-					_checked_register = (_checked_register + 1) % size_register_cfg;
+			// Check all Status/Error Flag Indicators (DIAG_STAT)
+			if (buffer.diag_stat != 0) {
+				perf_count(_bad_status_perf);
+				PX4_WARN("Error: DIAG_STAT: 0x%02x", buffer.diag_stat);
+				PrintErrorFlags(buffer.diag_stat);
+				return;
+			}
 
-				} else {
-					// register check failed, force reset
-					perf_count(_bad_register_perf);
-					Reset();
-				}
+			const float temperature = buffer.temp_out * 0.1f; // 1 LSB = 0.1°C
+			_px4_accel.set_temperature(temperature);
+			_px4_gyro.set_temperature(temperature);
+
+			// sensor frame is FLU, publish as FRD
+			float accel_x = buffer.x_accl_out;
+			float accel_y = -1.f * buffer.y_accl_out;
+			float accel_z = -1.f * buffer.z_accl_out;
+			float gyro_x = buffer.x_gyro_out;
+			float gyro_y = -1.f * buffer.y_gyro_out;
+			float gyro_z = -1.f * buffer.z_gyro_out;
+
+			// Group Delay with No Filtering: Accelerometer 1.57 ms
+			const uint64_t accel_group_delay_us = 1'570;
+			_px4_accel.update(timestamp_sample - accel_group_delay_us, accel_x, accel_y, accel_z);
+
+			// Group Delay with No Filtering:
+			//  Gyroscope (X-Axis) 1.51 ms
+			//  Gyroscope (Y-Axis) 1.51 ms
+			//  Gyroscope (Z-Axis) 1.29 ms
+			const uint64_t gyro_group_delay_us = (1'510 + 1'510 + 1'290) / 3;
+			_px4_gyro.update(timestamp_sample - gyro_group_delay_us, gyro_x, gyro_y, gyro_z);
+
+			if (_failure_count > 0) {
+				_failure_count--;
 			}
 		}
 
 		break;
 	}
 }
 
-bool ADIS16507::Configure()
+void ADIS16507::PrintErrorFlags(uint16_t flags)
 {
-	// first set and clear all configured register bits
-	for (const auto &reg_cfg : _register_cfg) {
-		RegisterSetAndClearBits(reg_cfg.reg, reg_cfg.set_bits, reg_cfg.clear_bits);
+	if (flags & (1 << 10)) {
+		PX4_WARN("Accelerometer failure");
```

Comment:
Be careful these can't spam.

Reply:
changed all potential spam prints to PX4_DEBUG